plugins/python/redisgears_python.c

```diff
@@ -1746,6 +1747,572 @@ static PyObject* atomicExit(PyObject *self, PyObject *args){
+    PyDAGRunner* pyDag = (PyDAGRunner*)self;
+    if(!PyObject_IsInstance((PyObject*)pyDag, (PyObject*)&PyDAGRunnerType)){
+        PyErr_SetString(GearsError, "Given argument is not of type DAGRunner");
+        return NULL;
+    }
+    if(PyTuple_Size(args) != 2){
+        PyErr_SetString(GearsError, "Wrong number of args to TensorSet op");
+        return NULL;
+    }
+    if(pyDag->dag == NULL){
+        PyErr_SetString(GearsError, "PyDAGRunner is invalid");
+        return NULL;
+    }
+    PyObject* tensorName = PyTuple_GetItem(args, 0);
+    if(!PyUnicode_Check(tensorName)){
+        PyErr_SetString(GearsError, "Tensor name argument must be a string");
+        return NULL;
+    }
+    const char* inputNameStr = PyUnicode_AsUTF8AndSize(tensorName, NULL);
+    PyTensor* pyt = (PyTensor*)PyTuple_GetItem(args, 1);
+    if(!PyObject_IsInstance((PyObject*)pyt, (PyObject*)&PyTensorType)){
+        PyErr_SetString(GearsError, "Given argument is not of type PyTensorType");
+        return NULL;
+    }
+    RedisAI_DAGAddTensorSet(pyDag->dag, inputNameStr, pyt->t);
+    Py_INCREF(self);
+    return self;
+}
+
+static PyObject* TensorGet(PyObject *self, PyObject *args) {
+    verifyRedisAILoaded();
+    if (RedisAI_DAGAddTensorGet == NULL) {
+        PyErr_SetString(GearsError, "DAG run is not supported in RedisAI version");
+        return NULL;
+    }
+    PyDAGRunner* pyDag = (PyDAGRunner*)self;
+    if(!PyObject_IsInstance((PyObject*)pyDag, (PyObject*)&PyDAGRunnerType)){
+        PyErr_SetString(GearsError, "Given argument is not of type DAGRunner");
+        return NULL;
+    }
+    if(PyTuple_Size(args) != 1){
+        PyErr_SetString(GearsError, "Wrong number of args to TensorSet op");
+        return NULL;
+    }
+    if(pyDag->dag == NULL){
+        PyErr_SetString(GearsError, "PyDAGRunner is invalid");
+        return NULL;
+    }
+    PyObject* tensorName = PyTuple_GetItem(args, 0);
+    if(!PyUnicode_Check(tensorName)){
+        PyErr_SetString(GearsError, "Tensor name argument must be a string");
+        return NULL;
+    }
+
+    const char* inputNameStr = PyUnicode_AsUTF8AndSize(tensorName, NULL);
+    RedisAI_DAGAddTensorGet(pyDag->dag, inputNameStr, NULL);
+    Py_INCREF(self);
+    return self;
+}
+
+static RAI_Model *_getModelFromKeyspace(const char *modelNameStr, RAI_Error *err) {
+    RedisModuleCtx *ctx = RedisModule_GetThreadSafeContext(NULL);
+    RedisGears_LockHanlderAcquire(ctx);
+    RedisModuleString* keyRedisStr = RedisModule_CreateString(ctx, modelNameStr, strlen(modelNameStr));
+    RedisModuleKey *key;
+    RAI_Model *model;
+    if(RedisAI_GetModelFromKeyspace(ctx, keyRedisStr, &key, &model, REDISMODULE_READ, err) != REDISMODULE_OK) {
+        RedisModule_FreeString(ctx, keyRedisStr);
+        RedisGears_LockHanlderRelease(ctx);
+        RedisModule_FreeThreadSafeContext(ctx);
+        PyErr_SetString(GearsError, RedisAI_GetError(err));
+        return NULL;
+    }
+
+    RedisModule_FreeString(ctx, keyRedisStr);
```

Comment:
Don't see where you close the key

Reply:
It's in RedisAI_GetModelFromKeyspace

Reply:
so why is the key for?

Reply:
You right, its redundant to create the key outside the function. Function signature is changed.